ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -169,32 +195,227 @@ module Json = struct
+    | _ ->
+        [
+          ("session_class", `Bool false)
+        ; ("session_login", `Bool false)
+        ; ("session_logout", `Bool false)
+        ]
+
+  let desc_of_msg msg ctor_fields =
+    let ctor =
+      if msg.msg_tag = FromObject Make then
+        Printf.sprintf " The constructor args are: %s (* = non-optional)."
+          ctor_fields
+      else
+        ""
+    in
+    match msg.msg_doc ^ ctor with
+    | "" ->
+        `Null
+    | desc ->
+        `String (String.trim desc)
+
+  let ctor_fields_of_obj obj =
+    Datamodel_utils.fields_of_obj obj
+    |> List.filter (function
+         | {qualifier= StaticRO | RW; _} ->
+             true
+         | _ ->
+             false
+         )
+    |> List.map (fun f ->
+           String.concat "_" f.full_name
+           ^ if f.default_value = None then "*" else ""
+       )
+    |> String.concat ", "
+
+  let messages_of_obj obj =
+    let ctor_fields = ctor_fields_of_obj obj in
+    let params_in_msg msg =
+      if msg.msg_session then
+        session_id :: msg.msg_params
+      else
+        msg.msg_params
+    in
+    List.map
+      (fun msg ->
+        let params = params_in_msg msg |> of_param obj.name in
+        let base_assoc_list =
+          [
+            ("method_name", `String msg.msg_name)
+          ; ("class_name", `String obj.name)
+          ; ("class_name_exported", `String (snake_to_camel obj.name))
+          ; ("method_name_exported", `String (snake_to_camel msg.msg_name))
+          ; ("description", desc_of_msg msg ctor_fields)
+          ; ("result", of_result obj msg)
+          ; ("params", params)
+          ; ("errors", of_errors msg.msg_errors)
+          ; ("has_error", `Bool (msg.msg_errors <> []))
+          ; ("async", `Bool msg.msg_async)
+          ]
+        in
+        `O (add_session_info obj.name msg.msg_name @ base_assoc_list)
+      )
+      obj.messages
+
   let xenapi objs =
     List.map
       (fun obj ->
+        let obj_name = snake_to_camel obj.name in
+        let name_internal = String.uncapitalize_ascii obj_name in
         let fields = Datamodel_utils.fields_of_obj obj in
         let types = List.map (fun field -> field.ty) fields in
         let modules =
           match obj.messages with [] -> `Null | _ -> modules_of_types types
         in
         let base_assoc_list =
           [
-            ("name", `String (snake_to_camel obj.name))
+            ("name", `String obj_name)
+          ; ("name_internal", `String name_internal)
           ; ("description", `String (String.trim obj.description))
           ; ( "fields"
             , `A (get_event_snapshot obj.name @ List.map of_field fields)
             )
           ; ("modules", modules)
+          ; ("messages", `A (messages_of_obj obj))
           ]
         in
         let assoc_list = base_assoc_list @ get_event_session_value obj.name in
         (String.lowercase_ascii obj.name, `O assoc_list)
       )
       objs
 
+  let of_api_message_or_error info =
```

Comment:
Would you please share more details why we change the naming for constant in `api_messages` and `api_errors`? Thank you!

Reply:
The go lint requires.

Reply:
Then I think it would be better to add this as commit message of the first commit, as it explains what the change is for.